pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -1177,12 +1195,12 @@ class Barrier(Operation):
 
     def __init__(self, only_visual=False, wires=Wires([]), do_queue=True, id=None):
         self.only_visual = only_visual
+        self.hyperparameters['only_visual'] = only_visual
         super().__init__(wires=wires, do_queue=do_queue, id=id)
 
-    # pylint: disable=unused-argument
-    def decomposition(self, wires):
-        # pylint: disable=unused-argument
-        return []
+    @staticmethod
+    def compute_decomposition(wires, **kwargs):
+        return tuple()
```

Comment:
Same question as above (and reminds me, you should return a tuple for identity decomposition too?)

Reply:
@albi3ro I think I am still quite wary about adding this change in this PR, since it is so different to our other methods (e.g., `diagonalizing_gates`, post-refactor, still returns a list).

Should we revert to a list here, and change _everything_ to a `tuple` in a separate PR, if that is the convention we want to set?

Reply:
Although I would argue this change is probably not high priority, and high effort to low impact.

Reply:
@josh146  I'm willing to revert this change for this PR, but I do think we should start using more tuples if we aren't needing to resize and edit them in most cases. Tuples are more efficient in both memory and time.

Reply:
In this case, I would probably prefer the change is reverted, to avoid introducing an inconsistency with the other methods that have already been refactored

Reply:
Agree both to using tuples in future, but not using them here to keep things simple!